src/istio/control/http/attributes_builder.cc

```diff
@@ -57,11 +58,46 @@ void AttributesBuilder::ExtractRequestHeaderAttributes(CheckData *check_data) {
   }
 }
 
-void AttributesBuilder::ExtractRequestAuthAttributes(CheckData *check_data) {
+void AttributesBuilder::ExtractAuthAttributes(CheckData *check_data) {
+  istio::authn::Result authn_result;
+  if (check_data->GetAuthenticationResult(&authn_result)) {
+    utils::AttributesBuilder builder(&request_->attributes);
+    if (!authn_result.principal().empty()) {
+      builder.AddString(AttributeName::kRequestAuthPrincipal,
+                        authn_result.principal());
+    }
+    if (!authn_result.peer_user().empty()) {
+      builder.AddString(AttributeName::kSourceUser, authn_result.peer_user());
+    }
+    if (authn_result.has_origin()) {
+      const auto &origin = authn_result.origin();
+      if (!origin.user().empty()) {
+        builder.AddString(AttributeName::kRequestAuthUser, origin.user());
+      }
+      if (!origin.audiences().empty()) {
```

Comment:
audiences is a repeated field, you need to check its _size().  I think

Reply:
empty also works for vector.